geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -5382,4 +5346,19 @@ public void registerPdxMetaData(Object instance) {
       throw new SerializationException("Serialization failed", e);
     }
   }
+
+  private void throwIfClient() {
+    if (isClient()) {
+      throw new UnsupportedOperationException("operation is not supported on a client cache");
+    }
+  }
+
+  private final InternalCacheForClientAccess cacheForClients =
+      new InternalCacheForClientAccess(this);
+
+  @Override
+  public InternalCache getCacheForClients() {
```

Comment:
The idea is for other teams to use this method to get a cache that will prevent accessing internal regions while processing messages on a server from a client.
The client checks needs to stay in GemFireCacheImpl since those are for caches created by ClientCacheFactory that live in the client jvm (not the server jvm).